src/job_consumer/utils_azure_queue.py

```diff
@@ -1,33 +1,55 @@
 import logging
+import os
 
-from azure.core.exceptions import ResourceExistsError
 from azure.identity import WorkloadIdentityCredential
 from azure.storage.queue import QueueClient, QueueServiceClient
 from django.conf import settings
 
 log = logging.getLogger(__name__)
 
 
-def get_queue_client():
-    if settings.AZURITE_QUEUE_CONNECTION_STRING:
-        queue_service_client = QueueServiceClient.from_connection_string(
-            settings.AZURITE_QUEUE_CONNECTION_STRING
-        )
-
-        # create queue for the first time
-        try:
-            queue_service_client.create_queue(settings.JOB_QUEUE_NAME)
-        except ResourceExistsError:
-            pass
-        
-        queue_client = queue_service_client.get_queue_client(settings.JOB_QUEUE_NAME)
-
-    else:
-        credentials = WorkloadIdentityCredential()
-        queue_client = QueueClient(
-            credential=credentials,
-            account_url=settings.QUEUE_ACCOUNT_URL,
-            queue_name=settings.JOB_QUEUE_NAME,
-        )
-
-    return queue_client
\ No newline at end of file
+class AzureQue:
+    ''' singleton Queue'''
+    _instance = None
+    _client = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(AzureQue, cls).__new__(cls)
+        return cls._instance
+
+    @classmethod
+    def _initialize_client(cls):
+        if cls._client is None:
+            # Maak de connectie met de Azure Queue
+            cls._client = cls._create_azure_queue_client()
```

Comment:
For me, this function does not do what it implies. When I look at where it is called, in get_queue_client, the function implies that it initializes the client every time. While that is the opposite of what it is doing. For me it would be clearer to move this function body inline into get_queue_client. What do you think?